### PR TITLE
testcases for tqdm_notebook and tnrange

### LINKED ISSUE #526: do you plan to write testcases for tqdm_notebook?
Well, using Javascript test tools like mocha, it is easy to write a Jupyter notebook, execute it, and check whether it has expected output.

If you think this is interesting, I can implement it.

### PR DESCRIPTION
**Pull request by chengs**
Resolve #526 

I write some assertions as testcases for tqdm_notebook into ``tqdm/tests/tests_notebook.ipynb``
And run it by
```
jupyter nbconvert --execute tqdm/tests/tests_notebook.ipynb
```

If all the assertions are passed, an HTML will be generated. 
If any of them cannot pass, jupyter nbconvert will raise an error.

P.S. I cannot find a way to combine ``nose`` and ``jupyter``. Maybe it is better to consider changing to ``py.test``

because of the following:
```
Nose has been in maintenance mode for the past several years and will likely cease without a new person/team to take over maintainership. New projects should consider using Nose2, py.test, or just plain unittest/unittest2.
```

@casperdcl, you can check https://travis-ci.org/chengs/tqdm/builds/426292378 for travis-CI output
screen shot 2018-09-09 at 4 45 42 pm
screen shot 2018-09-09 at 4 45 53 pm

### CODE REVIEW

**Comment by casperdcl** [comment]
Thanks, we probably need a new issue (if one doesn't exist already) for picking a new unit test framework

**Comment by chengs** [comment]
maybe in v5?

File: tox.ini [hunk 1]
```diff
@@ -89,3 +89,30 @@ deps =
...
+passenv = CI TRAVIS TRAVIS_*
+deps =
+    ipython<6
+    jupyter
+    pytest
+    nbval
+    pytest-cov
+    codecov
+commands =
+    pytest --cov=tqdm --nbval --current-env ./tqdm/tests/tests_notebook.ipynb
```
Comment (casperdcl) [inline tox.ini]: what is the `--current-env` for?

File: tqdm/tests/tests_notebook.ipynb [hunk 2]
```diff
@@ -0,0 +1,402 @@
...
+    "#     assert \"width='800px'\" in meta_widget or \"width=u'800px'\" in meta_widget    "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# TODO: reopen this after bugfix in python2\n",
```
Comment (casperdcl) [inline tqdm/tests/tests_notebook.ipynb]: what bugfix? Can these TODOs be addressed now?

Comment (chengs) [reply tqdm/tests/tests_notebook.ipynb]: Yes, the todos are addressed now. I uncommented them

**Comment by chengs** [comment]
@casperdcl, todos are addressed, now it can be merged into master.

**Comment by chengs** [comment]
@casperdcl ping

**Comment by casperdcl** [comment]
yes been über busy, will take a look soon hopefully

**Comment by chengs** [comment]
ping

**Pull request closed without merging**